File: src/main/java/com/itechart/tarasevi/logic/domain/Employee.java
//
// Source code recreated from a .class file by IntelliJ IDEA
// (powered by Fernflower decompiler)
//

package com.itechart.tarasevi.logic.domain;

import java.io.Serializable;
import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

public class Employee implements Serializable{
    private int id;
    private String firstName;
    private String lastName;
    private String patronymic;
    private LocalDate dateOfBirth;
    private String gender;
    private String nationality;
    private String familyStatus;
    private String webSite;
    private String email;
    private String workPlace;
    private Address address = new Address();
    private List<ContactPhone> phoneList = new ArrayList<>();
    private List<Attachment> attachmentList = new ArrayList<>();
    private String photoName;
    private Photo photo;

    public int getId() {
        return this.id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getFirstName() {
        return this.firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return this.lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public String getPatronymic() {
        return this.patronymic;
    }

    public void setPatronymic(String patronymic) {
        this.patronymic = patronymic;
    }

    public String getGender() {
        return this.gender;
    }

    public void setGender(String gender) {
        this.gender = gender;
    }

    public LocalDate getDateOfBirth() {
        return this.dateOfBirth;
    }

    public void setDateOfBirth(String dateOfBirth) {
        this.dateOfBirth = LocalDate.parse(dateOfBirth);
    }

    public String getNationality() {
        return this.nationality;
    }

    public void setNationality(String nationality) {
        this.nationality = nationality;
    }

    public String getFamilyStatus() {
        return this.familyStatus;
    }

    public void setFamilyStatus(String familyStatus) {
        this.familyStatus = familyStatus;
    }

    public String getWebSite() {
        return this.webSite;
    }

    public void setWebSite(String webSite) {
        this.webSite = webSite;
    }

    public String getEmail() {
        return this.email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getWorkPlace() {
        return this.workPlace;
    }

    public void setWorkPlace(String workPlace) {
        this.workPlace = workPlace;
    }

    public Address getAddress() {
        return this.address;
    }

    public void setAddress(Address address) {
        this.address = address;
    }

    public Photo getPhoto() {
        return photo;
    }

    public void setPhoto(Photo photo) {
        this.photo = photo;
    }

    public List<ContactPhone> getPhoneList() {
        return phoneList;
    }

    public void setPhoneList(List<ContactPhone> phoneList) {
        this.phoneList = phoneList;
    }

    public List<Attachment> getAttachmentList() {
        return attachmentList;
    }

    public void setAttachmentList(List<Attachment> attachmentList) {
        this.attachmentList = attachmentList;
    }

    public String getPhotoName() {
        return photoName;
    }

    public void setPhotoName(String photoName) {
        this.photoName = photoName;
    }

    @Override
    public String toString() {
        return "Employee{id=" + this.id + ", firstName=\'" + this.firstName + '\'' + ", lastName=\'" + this.lastName + '\'' + ", patronymic=\'" + this.patronymic + '\'' + ", dateOfBirth=" + this.dateOfBirth + ", gender=\'" + this.gender + '\'' + ", nationality=\'" + this.nationality + '\'' + ", familyStatus=\'" + this.familyStatus + '\'' + ", webSite=\'" + this.webSite + '\'' + ", email=\'" + this.email + '\'' + ", workPlace=\'" + this.workPlace + '\'' + ", address=" + this.address + '}';
    }


}

File: src/main/webapp/web/js/entity/employee.js
/**
 * Created by aefrd on 26.01.2017.
 */

function Employee() {
    this._id = undefined;
    this._firstName = undefined;
    this._lastName = undefined;
    this._firstName = undefined;
    this._lastName = undefined;
    this._patronymic = undefined;
    this._dateOfBirth = undefined;
    this._gender = undefined;
    this._nationality = undefined;
    this._familyStatus = undefined;
    this._webSite = undefined;
    this._email = undefined;
    this._workPlace = undefined;
    this._address = undefined;
    this._phoneList = [];
    this._attachmentList = [];
    this._photoName = undefined;

}

Employee.prototype = {
    set id(id) {
        this._id = id;
    },
    get id() {
        return this._id;
    },
    set firstName(firstName) {
        this._firstName = firstName;
    },
    get firstName() {
        return this._firstName;
    },
    set lastName(lastName) {
        this._lastName = lastName;
    },
    get lastName() {
        return this._lastName;
    },
    set  patronymic(patronymic) {
        this._patronymic = patronymic;
    },
    get  patronymic() {
        return this._patronymic;
    },
    set  dateOfBirth(dateOfBirth) {
        this._dateOfBirth = dateOfBirth;
    },
    get  dateOfBirth() {
        return this._dateOfBirth;
    },
    set gender(gender) {
        this._gender = gender;
    },
    get gender() {
        return this._gender;
    },
    set nationality(nationality) {
        this._nationality = nationality;
    },
    get nationality() {
        return this._nationality;
    },
    set familyStatus(familyStatus) {
        this._familyStatus = familyStatus;
    },
    get familyStatus() {
        return this._familyStatus;
    },
    set webSite(webSite) {
        this._webSite = webSite;
    },
    get webSite() {
        return this._webSite;
    },
    set email(email) {
        this._email = email;
    },
    get email() {
        return this._email;
    },
    set workPlace(workPlace) {
        this._workPlace = workPlace;
    },
    get workPlace() {
        return this._workPlace;
    },
    set address(address) {
        this._address = address;
    },
    get address() {
        return this._address;
    },
    set phoneList(phoneList) {
        this._phoneList = phoneList;
    },
    get phoneList() {
        return this._phoneList;
    },
    set attachmentList(attachmentList) {
        this._attachmentList = attachmentList;
    },
    get attachmentList() {
        return this._attachmentList;
    },
    set photoName(photoName) {
        this._photoName = photoName;
    },
    get photoName() {
        return this._photoName;
    }
};
